### PR TITLE
[DropdownButtonFormField] Add borderRadius property

### LINKED ISSUE #95926: [DropdownButtonFormField] Add `borderRadius` property
Flutter version: Channel stable, 2.8.0

[✓] Flutter (Channel stable, 2.8.0, on macOS 12.0.1 21A559 darwin-arm, locale zh-Hans-CN)
    • Flutter version 2.8.0 at /Users/rhyme/Documents/env/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision cf44000065 (3 weeks ago), 2021-12-08 14:06:50 -0800
    • Engine revision 40a99c5951
    • Dart version 2.15.0
    • Pub download mirror https://pub.flutter-io.cn
    • Flutter download mirror https://storage.flutter-io.cn

### PR DESCRIPTION
**Pull request by dheerajv09**
Adds borderRadius property  to DropdownButtonFormField.

fixes #95926

### CODE REVIEW

File: packages/flutter/test/material/dropdown_test.dart [hunk 1]
```diff
@@ -3678,4 +3678,56 @@ void main() {
 
     expect(tester.takeException(), null);
   });
+
+  group('BorderRadius property works properly for DropdownButtonFormField', () {
```
Comment (werainkhatri) [inline packages/flutter/test/material/dropdown_test.dart:3682]: Avoid using `group`s unless you really need it.

File: packages/flutter/test/material/dropdown_test.dart [hunk 2]
```diff
@@ -3678,4 +3678,56 @@ void main() {
...
+                  );
+              }).toList(),
+              onChanged: (_) { },
+            ),
+          ),
+        ),
+      );
+    }
+
+    testWidgets('When multiple items are available', (WidgetTester tester) async {
```
Comment (werainkhatri) [inline packages/flutter/test/material/dropdown_test.dart:3706]: Try to test this the same way `DropdownButton.borderRadius` is tested.

<details>
<summary>reference test</summary>

https://github.com/flutter/flutter/blob/39cdeb59621b144acd7424607c3bb80af20eb4a1/packages/flutter/test/material/dropdown_test.dart#L3528-L3573
</details>

**Review by werainkhatri — CHANGES_REQUESTED** [review]
Thank you for the contribution. This is a good addition to the API.

I just have a few suggestions.

**Review by werainkhatri — APPROVED** [review]
LGTM. Thank you for your contribution.

Although, this needs another reviewer's approval, as per our new commit policy.

File: packages/flutter/lib/src/material/dropdown.dart [hunk 3]
```diff
@@ -1615,6 +1616,7 @@ class DropdownButtonFormField<T> extends FormField<T> {
                      menuMaxHeight: menuMaxHeight,
                      enableFeedback: enableFeedback,
                      alignment: alignment,
+                     borderRadius: borderRadius,
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dropdown.dart:1619]: Can you add a comment here just like this one:
https://github.com/flutter/flutter/blob/da9e87021d9641460018c69455366dd87b621d68/packages/flutter/lib/src/material/dropdown.dart#L892

That will help us stay consistent both ways in the future. :)

File: packages/flutter/lib/src/material/dropdown.dart [hunk 4]
```diff
@@ -1540,6 +1540,7 @@ class DropdownButtonFormField<T> extends FormField<T> {
     double? menuMaxHeight,
     bool? enableFeedback,
     AlignmentGeometry alignment = AlignmentDirectional.centerStart,
+    BorderRadius? borderRadius,
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dropdown.dart:1543]: Oh whoops I meant here. Thanks!

File: packages/flutter/lib/src/material/dropdown.dart [hunk 5]
```diff
@@ -1540,6 +1540,9 @@ class DropdownButtonFormField<T> extends FormField<T> {
...
+    // When adding new arguments, consider adding similar arguments to
+    // DropdownButtonFormField.
```
Comment (Piinks) [inline packages/flutter/lib/src/material/dropdown.dart:1545]:
```suggestion
    // DropdownButton.
```

**Review by Piinks — APPROVED** [review]
Flutter_LGTM
Thank you for the contribution!

**Comment by dheerajv09** [comment]
:)

**Pull request merged**